### ISSUE TITLE
"osm_building_block_gen1()" function on planet import throws "invalid memory alloc request size"

### ISSUE THREAD

**Issue by enesaltinkaya** [openmaptiles/openmaptiles#1014]
During import-sql stage
```
CREATE MATERIALIZED VIEW osm_building_block_gen1 AS
SELECT *
FROM osm_building_block_gen1();  --line 82
```
results in this error
```
psql:building.sql:82: ERROR:  invalid memory alloc request size 3307581284
CONTEXT:  PL/pgSQL function osm_building_block_gen1() line 6 at FOR over SELECT rows
```

Planet import with DIFF_MODE=true

Ubuntu 20.04, i7-8700, 64GB, 2x nvme md raid 0, Postgresql 12.4, Postgis 3.0, default config
(Same error on, Ubuntu 18.04, Postgresql 10, Postgis 3.0, default config)

**Comment by TomPohys** [comment]
Hi @enesaltinkaya, please check your size of the Swap as I mention in https://github.com/openmaptiles/openmaptiles/issues/938#issuecomment-659266256

**Comment by enesaltinkaya** [comment]
Hi, sorry i forgot to mention it.
I set 70GB of swap.

```
      total        used        free      shared  buff/cache   available
Mem:           62Gi       585Mi       456Mi       8.3Gi        61Gi        53Gi
Swap:          69Gi       147Mi        69Gi
```

**Comment by TomPohys** [comment]
Maybe the planet will need a little bit more than 70GB of Swap. When I tested on France, I checked the graphs of how are filling the RAM and swapfile. Then I am extending the swapfile because RAM was too small.

**Comment by enesaltinkaya** [comment]
I see, thanks.

While "SELECT * FROM osm_building_block_gen1()" running, i never noticed swap being used for than a couple of hundred megabytes. This is from "free -h".

May I ask how you monitor RAM and swapfile usage?
Could it be that i need to make config change or something?

**Comment by TomPohys** [comment]
> May I ask how you monitor RAM and swapfile usage?

I used build-in GNOME System Monitor and htop

> Could it be that i need to make config change or something?

If you do not need buildings on z13 you can make modifications as I mention in https://github.com/openmaptiles/openmaptiles/issues/938#issuecomment-672252734

**Comment by enesaltinkaya** [comment]
Error occurs before reaching z13 related queries i think.

I'll take a look at postgres ram and swap usage.
Increase swap to 200GB and give it a try.

But ill have to do a fresh import.
Ill report in a couple of days.

**Comment by TomPohys** [comment]
`CREATE MATERIALIZED VIEW` is done in `make import-sql` step. `osm_building_block_gen1` materialized view is used for buildings in zoom level 13.

**Comment by enesaltinkaya** [comment]
I didn't know that. 
I'll try without building zoom 13.
Thanks a lot.

**Comment by TomPohys** [comment]
Hi @enesaltinkaya, any progress with this?

**Comment by enesaltinkaya** [comment]
Sorry, been busy. I had been trying to setup an auto updating tile server for about 10 days.
Results were expected. Private sector.

I wanted to experiment with more swap and zoom 13 buildings disabled.
But then I came across makina-maps, I noticed buildings zoom 13 is disabled there, so I used this one.
https://github.com/makina-maps/openmaptiles/tree/62b5146e483b40d25b6b1977115d4b7e363c2df8

Worked without issues, auto update and postserve is also working.
